### ISSUE TITLE
Stockholm not funny

### ISSUE THREAD

**Issue by mdehoon** [biopython/biopython#3747]
The `funny.sth` test file in `Tests/Stockholm` does not seem to be self-consistent.

For example on this line:
```
O83071/192-246                                                                  MTCRAQLIAVPRASSLAE..AIACAQKM....RVSRVPVYERS
```
the sequence length should be 248-(192 -1)= 57, but the actual sequence length is 37.

**Comment by peterjc** [comment]
Yes, was most likely deliberate although the commit history is not very explicit - I was using "funny" in the sense of strange, a deliberate atypical example for test purposes.

The sequence name can be anything, even if a suffix of "/start-end" is commonly used:

https://sonnhammer.sbc.su.se/Stockholm.html

We might issue a warning if the sequence name appears to follow the "/start-end" convention but is not consistent with the data?

**Comment by mdehoon** [comment]
Warnings are easy to miss, and are raised only once, so if `AlignIO.read` is called twice on different files that both have this problem, only the first one will show the warning. I think we should assume then that this was not intended, and always raise a `ValueError`.

**Comment by peterjc** [comment]
I don't think an exception is practical here, a warning or the current silent acceptance seems safer.

Making such files is very easy (in any file format). Take any "nice" alignment with records named like this, and do some column slicing. Biopython will write it with the original identifiers and their original /start-end suffix. When I was doing this to crop to a domain only, as I recall, I would typically drop or adjust the suffixes in code.

Looking back I think I made the wrong choice for ``.name`` versus ``.id`` when recognising the "identifier/start-end" pattern was being used:

https://github.com/biopython/biopython/blob/biopython-179/Bio/AlignIO/StockholmIO.py#L500

Currently ".id" is the full string, and ".name" has the suffix removed. Perhaps both should have the suffix removed (indeed, having two attributes here has long struck me as unhelpful in the ``SeqRecord`` design).

**Comment by peterjc** [comment]
That was probably accidental on my part.
